### PR TITLE
Firmware cleanup and integration

### PR DESCRIPTION
**Pull request by micronova-jb**
Added the source files for the preamp firmware to the GitHub repository. I'm not sure why i2c_regs.md has extra trailing cells in the lines that 'span.' I can't seem to find a solution for this.

### CODE REVIEW

File: fw/preamp/src/channel.c [hunk 1]
```diff
@@ -0,0 +1,195 @@
+/*
+ * AmpliPi Home Audio
+ * Copyright (C) 2021 MicroNova LLC
+ *
```
Comment (linknum23) [inline fw/preamp/src/channel.c:4]: We need to give a little info about each of these modules, first stab:
Configure each of the preamp controller's 6 channels. 

Each channel has several different configurable properties:
- Attenuation
- Standby
- Mute
- Audio Source

File: fw/preamp/src/channel.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+ * AmpliPi Home Audio
+ * Copyright (C) 2021 MicroNova LLC
+ *
```
Comment (linknum23) [inline fw/preamp/src/channel.h:4]: same

File: fw/preamp/src/syscalls.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
+/**
```
Comment (linknum23) [inline fw/preamp/src/syscalls.c:1]: Is this file needed? Try removing it

File: fw/preamp/src/ports.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	case 'B':
+		return GPIOB;
+	case 'C':
+		return GPIOC;
+	case 'D':
+		return GPIOD;
+	case 'F':
+		return GPIOF;
+	default:
+		while(1);
```
Comment (linknum23) [inline fw/preamp/src/ports.c:37]: This should be removed in production. I would rather return an invalid port then halt.

File: fw/preamp/src/port_defs.c [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "port_defs.h"
+
```
Comment (linknum23) [inline fw/preamp/src/port_defs.c:22]: Add comment:
// Enable pin mapping for each channel's 4 sources
// Each channel can enable all or none of its four sources. This firmware currently only allows one to be enabled at time.

**Review by linknum23 — CHANGES_REQUESTED** [review]
Looking good just a couple of changes...

**Comment by linknum23** [comment]
Also just a reminder, once all of these changes have been made. Please make a build of the firmware from a freshly checked out version of the amplipi repo (on this branch) and make sure the build succeeds!

File: fw/preamp/src/syscalls.c (hunk 3 above)
Comment (micronova-jb) [reply fw/preamp/src/syscalls.c:1]: I missed this one! Compiling without it threw no errors, and the binary didn't change at all

**Review by linknum23 — COMMENTED** [review]
Looking good! Let me know when this FW has been built, tested and verified from the AmpliPi repo and I will approve it.

**Review by linknum23 — COMMENTED** [review]
Where is the build procedure documented?

**Review by linknum23 — APPROVED** [review]
